.github/workflows/main.yml

```diff
@@ -50,43 +51,33 @@ jobs:
       name: Upload Mac pkg
       uses: actions/upload-artifact@v2
       with:
+        name: macos-package
         path: ./build/jpackage/*.pkg
-        name: ${{env.MAC_SAVEAS}}
+        if-no-files-found: error
         retention-days: 3
-      env:
-        MAC_SAVEAS: ${{env.APP_NAME}}-${{env.APP_VERSION}}-MacOS.pkg
     - if: ${{ matrix.os == 'windows-latest' }}
-      name: Upload Windows msi
+      name: Upload Windows MSI
       uses: actions/upload-artifact@v2
       with:
+        name: windows-package
         path: ./build/jpackage/*.msi
-        name: ${{env.WIN_SAVEAS}}
+        if-no-files-found: error
         retention-days: 3
-      env:
-        WIN_SAVEAS: ${{env.APP_NAME}}-${{env.APP_VERSION}}-win.msi
-    strategy:
-      matrix:
-        os: [ macos-latest, windows-latest ]
   upload:
-    name: Create Release
+    name: Create release
```

Comment:
Primary question from my side is whether this stage is unnecessary. As the artifacts need some signing steps and are hosted on https://downloads.glencoesoftware.com/public/NGFF-Converter, we made the decision to remove the former GitHub releases with the unsigned artifacts to prevent confusion

As far as the release process goes, it should be sufficient to download the artifacts associated with a tag build (within the retention period) and start the code signing. Do we need the create of a draft release at all?

Reply:
Short answer is that I don't know so I kept it as is. The biggest problem with not creating a release as far as the downstream code signing repositories go is that it adds complexity. In order to download artifacts from a GitHub action workflow execution some URL mangling is required and you must be authenticated.

Reply:
Create Release was added before we'd found that signing needs to be done outside the main workflow. I've no objections to removing/replacing it. There should be a means to either grab the artifacts from the actions workflow, or otherwise trigger signing once this action completes and send the signed artifacts back automatically.

Reply:
> In order to download artifacts from a GitHub action workflow execution some URL mangling is required and you must be authenticated.

:+1: I had not considered the authentication requirement and I agree it unnecessarily adds complexity to the downstream code signing process. Happy to go with the draft release approach for this round which should have the advantage of generating pre-computable URLs. Possibly the only minor suggestion would be to use the `body:` field in the gh release action with an explicit about the fact these are unsigned artifacts.